scipy/stats/tests/test_morestats.py

```diff
@@ -2975,90 +2980,96 @@ def test_directional_stats_correctness(self):
         circular_var = stats.circvar(testdata)
-        assert_allclose(directional_var, circular_var)
+        xp_assert_close(directional_var, circular_var)
 
-    def test_directional_mean_higher_dim(self):
+    def test_directional_mean_higher_dim(self, xp):
         # test that directional_stats works for higher dimensions
         # here a 4D array is reduced over axis = 2
         data = np.array([[0.8660254, 0.5, 0.],
                          [0.8660254, -0.5, 0.]])
-        full_array = np.tile(data, (2, 2, 2, 1))
-        expected = np.array([[[1., 0., 0.],
-                              [1., 0., 0.]],
-                             [[1., 0., 0.],
-                              [1., 0., 0.]]])
+        full_array = xp.asarray(np.tile(data, (2, 2, 2, 1)))
+        expected = xp.asarray([[[1., 0., 0.],
+                                [1., 0., 0.]],
+                               [[1., 0., 0.],
+                                [1., 0., 0.]]],
+                              dtype=xp.float64)
         dirstats = stats.directional_stats(full_array, axis=2)
-        assert_allclose(expected, dirstats.mean_direction)
+        xp_assert_close(dirstats.mean_direction, expected)
 
-    def test_directional_stats_list_ndarray_input(self):
+    def test_directional_stats_list_ndarray_input(self, xp):
         # test that list and numpy array inputs yield same results
         data = [[0.8660254, 0.5, 0.], [0.8660254, -0.5, 0]]
-        data_array = np.asarray(data)
-        res = stats.directional_stats(data)
-        ref = stats.directional_stats(data_array)
-        assert_allclose(res.mean_direction, ref.mean_direction)
-        assert_allclose(res.mean_resultant_length,
-                        res.mean_resultant_length)
-
-    def test_directional_stats_1d_error(self):
+        data_array = xp.asarray(data, dtype=xp.float64)
+        ref = stats.directional_stats(data)
+        res = stats.directional_stats(data_array)
+        xp_assert_close(res.mean_direction,
+                        xp.asarray(ref.mean_direction))
+        xp_assert_close(res.mean_resultant_length,
+                        xp.asarray(res.mean_resultant_length))
+
+    def test_directional_stats_1d_error(self, xp):
         # test that one-dimensional data raises ValueError
-        data = np.ones((5, ))
+        data = xp.ones((5, ))
         message = (r"samples must at least be two-dimensional. "
                    r"Instead samples has shape: (5,)")
         with pytest.raises(ValueError, match=re.escape(message)):
             stats.directional_stats(data)
 
-    def test_directional_stats_normalize(self):
+    def test_directional_stats_normalize(self, xp):
         # test that directional stats calculations yield same results
         # for unnormalized input with normalize=True and normalized
         # input with normalize=False
         data = np.array([[0.8660254, 0.5, 0.],
                          [1.7320508, -1., 0.]])
```

Comment:
Just thought I'd note that implicitly this test is `float64`-only, but that's OK with me. If you feel like `parametrize`ing over `float32` and `float64` to firmly put float dtype concerns to rest, that would be awesome.